File: test/m68k/integration/fib.zig
const std = @import("std");

pub const Input = extern struct {
    to: u32,
};

pub const Output = extern struct {
    result: u32,
};

fn fib(n: u32) u32 {
    var a: u32 = 0;
    var b: u32 = 1;
    for (0..n) |_| {
        const next = a +% b;
        a = b;
        b = next;
    }
    return a;
}

pub export fn run(input: *const Input, output: *Output) void {
    output.result = fib(input.to);
}

File: test/m68k/integration/fib.c
#include "fib.h"

static uint16_t fib(uint16_t n)
{
    uint16_t a = 0;
    uint16_t b = 1;
    while (n) {
        const uint16_t next = a + b;
        a = b;
        b = next;
        --n;
    }
    return a;
}

void run(const struct Input* input, struct Output* output)
{
    output->result = fib(input->to);
}
